packages/node_modules/@webex/webex-core/src/lib/services/service-catalogs.js

```diff
@@ -41,6 +41,72 @@ export default class ServiceCatalogs {
     this.hosts = [];
   }
 
+  /**
+   * Filter the {@link ServiceCatalogs#hosts} array against their active states.
+   *
+   * @private
+   * @memberof ServiceCatalogs
+   * @param {Object} pto - The parameter transfer object.
+   * @property {boolean} [pto.active] - Filter to ignore service state.
+   * @returns {Array<ServiceHost>} - The filtered host array.
+   */
+  filterActive({active = true} = {}) {
+    // Filter the host array if the active requirement is true.
+    return (active === true) ?
+      this.hosts.filter((host) => host.active) :
+      [...this.hosts];
+  }
+
+  /**
+   * Filter the {@link ServiceCatalogs#hosts} array against their assigned
+   * catalog values.
+   *
+   * @private
+   * @memberof ServiceCatalogs
+   * @param {Object} pto - The parameter transfer object.
+   * @property {Array<string> | string} [pto.catalog] - Catalogs to filter.
+   * @returns {Array<ServiceHost>} - The filtered host array.
+   */
+  filterCatalog({catalog} = {}) {
+    // Generate usable catalog names based on the provided catalog array.
+    let catalogs = [];
+
+    // Validate if the catalog param exists.
+    if (catalog) {
```

Comment:
You could do this the same way you do the `filterCluster` instead of creating the empty `catalogs` array.

Reply:
Updated to make these two match.